pkg/smi/traffic_target.go

```diff
@@ -0,0 +1,168 @@
+	if check.cfg.IsPermissiveTrafficPolicyMode() {
+		return outcomes.DiagnosticOutcome{LongDiagnostics: fmt.Sprintf("OSM is in permissive traffic policy modes -- all meshed pods can communicate and SMI access policies are not applicable")}
+	}
+	trafficTargets, err := check.accessClient.AccessV1alpha3().TrafficTargets(check.dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return outcomes.FailedOutcome{Error: err}
+	}
+	var matchingTargetNames []string
+	for _, trafficTarget := range trafficTargets.Items {
+		if doesTargetMatchPods(trafficTarget.Spec, check.srcPod, check.dstPod) {
+			matchingTargetNames = append(matchingTargetNames, trafficTarget.Name)
+		}
+	}
+	if len(matchingTargetNames) > 0 {
+		return outcomes.DiagnosticOutcome{LongDiagnostics: fmt.Sprintf("Pod '%s/%s' is allowed to communicate to pod '%s/%s' via SMI TrafficTarget policy/policies %s\n",
+			check.srcPod.Namespace, check.srcPod.Name, check.dstPod.Namespace, check.dstPod.Name, strings.Join(matchingTargetNames, ", "))}
+	}
+	return outcomes.DiagnosticOutcome{LongDiagnostics: fmt.Sprintf("Pod '%s/%s' is not allowed to communicate to pod '%s/%s' via any SMI TrafficTarget policy\n",
+		check.srcPod.Namespace, check.srcPod.Name, check.dstPod.Namespace, check.dstPod.Name)}
+}
+
+// Suggestion implements common.Runnable
+func (check TrafficTargetCheck) Suggestion() string {
+	return fmt.Sprintf("Check that source and desintation pod are referred to in a TrafficTarget. To get relevant TrafficTargets, use: \"kubectl get traffictarget -n %s -o yaml\"", check.dstPod.Namespace)
+}
+
+// FixIt implements common.Runnable
+func (check TrafficTargetCheck) FixIt() error {
+	panic("implement me")
+}
+
+func doesTargetMatchPods(spec access.TrafficTargetSpec, srcPod *corev1.Pod, dstPod *corev1.Pod) bool {
+	// Map traffic targets to the given pods
+	if !cli.DoesTargetRefDstPod(spec, dstPod) {
+		return false
+	}
+	// The TrafficTarget destination is associated to 'dstPod', check if 'srcPod` is an allowed source to this destination
+	if cli.DoesTargetRefSrcPod(spec, srcPod) {
+		return true
+	}
+	return false
+}
+
+// Verify interface compliance
+var _ common.Runnable = (*RoutesValidityCheck)(nil)
+
+// TrafficTargetCheck implements common.Runnable
+type RoutesValidityCheck struct {
+	cfg          configurator.Configurator
+	srcPod       *corev1.Pod
+	dstPod       *corev1.Pod
+	accessClient smiAccessClient.Interface
+}
+
+// IsInTrafficTarget checks whether the src and dest pods are referenced as src and dest in a TrafficTarget (in that order)
+func AreTrafficRoutesValid(osmConfigurator configurator.Configurator, srcPod *corev1.Pod, dstPod *corev1.Pod, smiAccessClient smiAccessClient.Interface) RoutesValidityCheck {
+	return RoutesValidityCheck{
+		cfg:          osmConfigurator,
+		srcPod:       srcPod,
+		dstPod:       dstPod,
+		accessClient: smiAccessClient,
+	}
+}
+
+const (
+	HTTPRouteGroupKind = "HTTPRouteGroup"
+	TCPRouteKind       = "TCPRoute"
+)
```

Comment:
is there anywhere in osm that already has these defined as supported routes?

Reply:
I used this:
```bash
$ grep -Iir ' "HTTPRouteGroup"' $(find . -name '*.go') --color | grep '='                                                                                                                                                                                                                                                                                                                                                                                                                                    08/24/21  2:45 PM
./pkg/validator/validators.go:          if m.Kind != "HTTPRouteGroup" {
./pkg/catalog/traffictarget.go: httpRouteGroupKind = "HTTPRouteGroup"
```

So I see these here: https://github.com/openservicemesh/osm/blob/c05c35aea5d10b11237975c901bfe6ed3a0f8eca/pkg/catalog/traffictarget.go#L14-L23


But I'm not sure that it is worth another cross dependency just to bring in these constants.